presto-docs/src/main/sphinx/functions/array.rst

```diff
@@ -62,6 +62,25 @@ Array Functions
     Sorts and returns the array ``x``. The elements of ``x`` must be orderable.
     Null elements will be placed at the end of the returned array.
 
+.. function:: array_sort(array<T>, function<T,T,int>) -> array<T>
+
+    Sorts and returns the ``array`` based on the given comparator ``function``. The comparator will take
+    two non-null arguments representing two non-null elements of the ``array``. It returns -1, 0, or 1
+    as the first non-null element is less than, equal to, or greater than the second non-null element.
+    Null elements will be ignored and placed at the end of the returned array. ::
```

Comment:
Don't ignore nulls. With a custom comparator, it makes sense to let the comparator choose. What do you think? If you agree, an example that does null first and another that does null last will be helpful. Please also test it.

Reply:
Also document what happens when
- comparator returns value other than -1, 0 or 1 (i understand this is an error and fails the query)
- comparator returns `null` (does it make the whole `array_sort` to return `null` as well? Or, does it fail the query?)

Reply:
@haozhun I agree with you on letting the users have more controls with the comparator function on nulls. Let me fix it then.